basiclti/basiclti-impl/src/java/org/sakaiproject/basiclti/impl/BasicLTISecurityServiceImpl.java

```diff
@@ -262,15 +262,22 @@ private void doSplash(HttpServletRequest req, HttpServletResponse res, String sp
 		login_hint	required, Hint to the Authorization Server about the login
 					identifier the End-User might use to log in (if necessary).
 	*/
-	private void redirectOIDC(HttpServletRequest req, HttpServletResponse res, String oidc_endpoint, ResourceLoader rb)
+	private void redirectOIDC(HttpServletRequest req, HttpServletResponse res, Map<String, Object> tool, String oidc_endpoint, ResourceLoader rb)
 	{
 		// req.getRequestURL()=http://localhost:8080/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
 		// req.getRequestURI()=/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
 		String login_hint = req.getRequestURI().toString();
-		// TODO: Encrypt the login hint
+		String query_string = req.getQueryString();
+		if ( StringUtils.isNotEmpty(query_string)) {
+			login_hint = login_hint + "?" + query_string;
```

Comment:
Seems more clear to show your concatenation:

login_hint += "?" + query_string;